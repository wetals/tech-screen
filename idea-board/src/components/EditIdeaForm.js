import React from "react";
import { MAX_COUNT } from "./shared";

const EditIdeaForm = props => {
  const handleChange = ({ currentTarget }) => {
    if (
      currentTarget.name === "description" &&
      currentTarget.value.length > MAX_COUNT
    ) {
      return;
    }
    const updatedIdea = {
      ...props.idea,
      [currentTarget.name]: currentTarget.value,
      lastUpdated: Date.now(),
      active: true
    };
    props.handleUpdateIdea(props.index, updatedIdea);
  };

  const updateActiveStatus = event => {
    const updatedIdea = {
      ...props.idea,
      active: event.type === "focus"
    };
    props.handleUpdateIdea(props.index, updatedIdea);
  };

  const formatDate = date => {
    const options = {
      year: "numeric",
      month: "short",
      day: "numeric"
    };
    const ideaDate = new Date(date);

    return ideaDate.toLocaleDateString([], options);
  };

  const { title, description, active, lastUpdated } = props.idea;
  const className = `idea edit-mode ${active ? "focus" : ""}`;
  let remainingChar = MAX_COUNT - description.length;

  return (
    <div
      className={className}
      onBlur={updateActiveStatus}
      onFocus={updateActiveStatus}
    >
      <div className="field title">
        <input type="text" name="title" onChange={handleChange} value={title} />
      </div>

      <div className="field description">
        <textarea
          name="description"
          onChange={handleChange}
          value={description}
        />
        {remainingChar < 20 &&
          active && (
            <span className="counter">
              {`${remainingChar} characters remaining`}
            </span>
          )}
      </div>

      <div className="action-button">
        <button onClick={() => props.handleDeleteIdea(props.index)}>
          Remove
        </button>

        <div className="idea_date">
          <span>{formatDate(lastUpdated)}</span>
        </div>
      </div>
    </div>
  );
};

export default EditIdeaForm;
