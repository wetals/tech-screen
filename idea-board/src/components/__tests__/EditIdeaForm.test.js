import React from "react";
import { shallow } from "enzyme";

import EditIdeaForm from "../EditIdeaForm";
import renderer from "react-test-renderer";

const testData = {
  index: 0,
  idea: {
    title: "test",
    description: "test",
    lastUpdated: 1537764753873,
    active: false
  },
  handleUpdateIdea: jest.fn(),
  handleDeleteIdea: jest.fn()
};

describe("EditIdeaForm component", () => {
  it("renders properly", () => {
    const wrapper = shallow(<EditIdeaForm {...testData} />);
  });

  test("renders snapshot properly", () => {
    const tree = renderer.create(<EditIdeaForm {...testData} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
