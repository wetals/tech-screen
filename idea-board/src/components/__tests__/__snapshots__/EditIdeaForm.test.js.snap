// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditIdeaForm component renders snapshot properly 1`] = `
<div
  className="idea edit-mode "
  onBlur={[Function]}
  onFocus={[Function]}
>
  <div
    className="field title"
  >
    <input
      name="title"
      onChange={[Function]}
      type="text"
      value="test"
    />
  </div>
  <div
    className="field description"
  >
    <textarea
      name="description"
      onChange={[Function]}
      value="test"
    />
  </div>
  <div
    className="action-button"
  >
    <button
      onClick={[Function]}
    >
      Remove
    </button>
    <div
      className="idea_date"
    >
      <span>
        2018 M09 24
      </span>
    </div>
  </div>
</div>
`;
